iModelJsNodeAddon/api_package/ts/src/test/DgnDb.test.ts

```diff
@@ -510,4 +510,202 @@ describe("basic tests", () => {
       }],
     });
   });
+
+  it("testConvertEC2XmlSchemas", async () => {
+    const ec2SchemaXml = `<?xml version="1.0" encoding="UTF-8"?>
+      <ECSchema schemaName="TestSchema" version="1.0" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.2.0">
+        <ECSchemaReference name="RefSchema" version="01.00" prefix="rs" />
+        <ECClass typeName="TestEntityClass" isDomainClass="true" />
+      </ECSchema>`;
+
+    const ec2RefSchema = `<?xml version="1.0" encoding="UTF-8"?>
+      <ECSchema schemaName="RefSchema" nameSpacePrefix="rs" version="1.0" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.2.0">
+        <ECClass typeName="TestStructClass" isStruct="true" />
+      </ECSchema>`;
+
+    const ec3Schemas: string[] = iModelJsNative.SchemaUtility.convertEC2XmlSchemas([ec2SchemaXml, ec2RefSchema]);
+    assert.equal(ec3Schemas.length, 2);
+    // converted EC3 schemas are always sorted in dependancy order: RefSchema-> TestSchema
+    const ec3RefSchema = ec3Schemas[0];
+    const ec3SchemaXml = ec3Schemas[1];
+
+    assert.isTrue(ec3RefSchema.includes("http://www.bentley.com/schemas/Bentley.ECXML.3.2"));
+    assert.isTrue(ec3SchemaXml.includes("http://www.bentley.com/schemas/Bentley.ECXML.3.2"));
+  });
+
+  it("testConvertEC2XmlSchemas enumeration in ref schema", async () => {

```

Comment:
Adding comment to track. Like I mentioned in teams, I think it would be a good test to put schema things in there which have reserved names, so the converter renames them, and then later the test verifies the things have been renamed (the reserved name thing should be gone). This should be a reliable method to verify the conversion worked successfully.